lib/core/src/lib/form/components/widgets/core/form-field.model.spec.ts

```diff
@@ -740,96 +752,112 @@ describe('FormFieldModel', () => {
-            ]
+    describe('dropdown field of rest type', () => {
+        it('should assign rest properties properly', () => {
+            const form = new FormModel();
+            const field = new FormFieldModel(form, {
+                fieldType: 'HeaderFieldtype',
+                id: 'dropdown_field',
+                name: 'header',
+                type: FormFieldTypes.DROPDOWN,
+                value: '',
+                required: false,
+                readOnly: true,
+                optionType: 'rest',
+                restUrl: 'fake-url',
+                restIdProperty: 'fake-id-property',
+                restLabelProperty: 'fake-label-property',
+                restResponsePath: 'fake-response-path',
+                options: []
+            });
+            field.updateForm();
+
+            expect(field.value).toEqual('');
+            expect(form.values['dropdown_field']).toEqual(null);
+            expect(field.options).toEqual([]);
+            expect(field.restUrl).toEqual('fake-url');
+            expect(field.restIdProperty).toEqual('fake-id-property');
+            expect(field.restLabelProperty).toEqual('fake-label-property');
+            expect(field.restResponsePath).toEqual('fake-response-path');
         });
-        field.updateForm();
-        expect(form.values['dropdown_field'].id).toEqual('opt1');
-        expect(form.values['dropdown_field'].name).toEqual('Option 1');
-    });
 
-    it('dropdown field type should be formatted on id and name properties if rest properties are not set', () => {
-        const form = new FormModel();
-        const field = new FormFieldModel(form, {
-            fieldType: 'HeaderFieldtype',
-            id: 'dropdown_field',
-            name: 'header',
-            type: FormFieldTypes.DROPDOWN,
-            value: 'opt1',
-            required: false,
-            readOnly: true,
-            restUrl: 'fake-url-just-to-show',
-            optionType: 'rest',
-            options: [
-                { id: 'opt1', name: 'Option 1' },
-                { id: 'opt2', name: 'Option 2' }
-            ]
+        it('should skip list of static options and assign empty array instead', () => {
+            const form = new FormModel();
+            const field = new FormFieldModel(form, {
+                fieldType: 'HeaderFieldtype',
+                id: 'dropdown_field',
+                name: 'header',
+                type: FormFieldTypes.DROPDOWN,
+                value: 'delayed-rest-option-id',
+                required: false,
+                readOnly: true,
+                optionType: 'rest',
+                restUrl: 'fake-url',
+                options: [
+                    { id: 'opt1', name: 'Option 1' },
+                    { id: 'opt2', name: 'Option 2' }
+                ]
+            });
+            field.updateForm();
+
+            expect(field.value).toEqual('delayed-rest-option-id');
+            expect(form.values['dropdown_field']).toEqual(undefined);
```

Comment:
Just noticed that I have here **undefined** check but in other cases **null**, will check it

Reply:
Solved